src/utils/data/getPropertyValue.js

```diff
@@ -0,0 +1,17 @@
+/*
+ * @flow
+ */
+
+import { Map, getIn } from 'immutable';
+import { Models } from 'lattice';
+
+const { FQN } = Models;
+
+export default function getPropertyValue(
+  entity :Map | Object,
+  fqn :FQN | string,
+  defaultValue ? :string = ''
+) :any {
+
+  return getIn(entity, [fqn, 0], defaultValue);
+}
```

Comment:
I don't think it's a good idea to hardcode the `0` index here. I think the following would be a bit more flexible AND removes the need for `getPropertyMultiplicityValue` since it accomplishes the same thing:

```
export default function getPropertyValue(
  entity :Map | Object,
  key :FQN | string | Array<FQN | string | number>,
  fallback ?:boolean | number | string = '',
) :boolean | number | string | Array<boolean | number | string> {

  if (isNonEmptyArray(key)) {
    return getIn(entity, key, fallback);
  }
  else if (isNonEmptyString(key)) {
    return get(entity, key, fallback);
  }

  return fallback;
}
```

By the way, we know that property values can only be booleans, numbers, or strings, so instead of `:any` we can type it as `:boolean | number | string`.

Why do you want to pass a fallback / default value? Do you think the following would be ok (just defaulting to `undefined`)?

```
export default function getPropertyValue(
  entity :Map | Object,
  key :FQN | string | Array<FQN | string | number>,
) :?boolean | number | string | Array<boolean | number | string> {

  if (isNonEmptyArray(key)) {
    return getIn(entity, key);
  }
  else if (isNonEmptyString(key)) {
    return get(entity, key);
  }

  return undefined;
}

const value = getPropertyValue(...) || 'defaultValue';
```

Reply:
I like this idea much better, thank you! Definitely seems better to have one function for both single values and property multiplicity values.

As for the fallback value, the reason I've always included it in utils like this is to be able to pass the retrieved value directly to the UI component and not have to include an extra line for what to display if the value is undefined. I usually pass '----' as a fallback. But it's not a problem to eliminate the fallback if you think other people wouldn't use it in the same way.

Reply:
I updated this to use your first suggested rewrite! This is second if block though, to allow for both string and FQN keys:
```
if (isNonEmptyString(key) || isObject(key)) {
  return get(entity, key, fallback);
}